### ISSUE TITLE
Automatic encoding detection for template files fails for single german umlaut

### ISSUE THREAD

**Issue by timorieber** [invoice-x/invoice2data#445]
In some cases `chardet.detect` fails to detect the correct template encoding. This leads to wrong `InvoiceTemplate` objects and wrong parsing results in the end.

Example template:
```yaml
issuer: Basic Test
keywords:
  - Basic Test
fields:
  date:
    parser: regex
    regex: Issue date:\s*(\d{4}-\d{2}-\d{2})
    type: date
  invoice_number:
    parser: regex
    regex: Invoice number:\s*([\d/]+)
  amount:I have implemented a new `option` to specify the actual template encoding and extended the loader mechanism
    parser: regex
    regex: Total:\s*(\d+\.\d\d)
    type: float
  single_specialchar:
    parser: static
    value: ä
```

If I add another umlaut the detection works and identifies as `utf-8`. As a quick workaround one could add another umlaut to an unused static value.

**Comment by rmilecki** [comment]
I agree that `chardet` is unreliable. I had experienced problems with UTF-8 encoded tpl files containing Polish characters. Encoding was not correct detected as UTF-8.

Another user had similar problem:  `chardet` was incorrectly detecting encoding of tpl file containing `€` char, see: https://github.com/invoice-x/invoice2data/issues/368

---

All `invoice2data` internal templates include encoding info in a special comment line like:
```
# -*- coding: utf-8 -*-
```
Is that actually something respected by `invoice2data` or `chardet`? I don't see any relevant code in any of these projects.

That comment looks like Python syntax copied into YAML syntax and it appears to be useless.

---

As for YAML itself, all processors must support UTF-8, see: https://yaml.org/spec/1.2.2/#52-character-encodings

I'm wondering if we need all that unreliable detecting code at all. Could we just treat all YAML files as UTF-8?

**Comment by timorieber** [comment]
When I encountered this problem I did some tests regarding the special comment line in templates and can confirm it has no effect. The exact value of `confidence` in the chardet result did not change at all. Therefore it is more misleading then helpful and should be removed from internal templates.

I also support the idea to remove encoding detection completely in favour of treating all templates as `utf-8`. But this should be another step in the future, because that requires more thoughts and work, e.g. migrating existing templates...

**Comment by rmilecki** [comment]
> that requires more thoughts and work, e.g. migrating existing templates

Luckily for us all internal templates actually use UTF-8 encoding.
I also think there is a low risk of external templates using other encodings and depending on `chardet` which clearly isn't reliable.

**Comment by rmilecki** [comment]
Solved by 6ab31047386a ("Treat all YAML templates files as UTF-8 encoded")
